core/encoding/json/marshal.odin

```diff
@@ -17,25 +17,31 @@ Marshal_Error :: union #shared_nil {
 	io.Error,
 }
 
-marshal :: proc(v: any, allocator := context.allocator) -> (data: []byte, err: Marshal_Error) {
+Pretty :: struct {
+	apply: bool,
+	indentation: int,
+}
+
+marshal :: proc(v: any, pretty_print := false, allocator := context.allocator) -> (data: []byte, err: Marshal_Error) {
 	b := strings.make_builder(allocator)
 	defer if err != nil {
 		strings.destroy_builder(&b)
 	}
 
-	marshal_to_builder(&b, v) or_return
+	pretty := Pretty { apply = pretty_print }
+	marshal_to_builder(&b, v, &pretty) or_return
 	
 	if len(b.buf) != 0 {
 		data = b.buf[:]
 	}
 	return data, nil
 }
 
-marshal_to_builder :: proc(b: ^strings.Builder, v: any) -> Marshal_Error {
-	return marshal_to_writer(strings.to_writer(b), v)
+marshal_to_builder :: proc(b: ^strings.Builder, v: any, pretty: ^Pretty) -> Marshal_Error {
+	return marshal_to_writer(strings.to_writer(b), v, pretty)
 }
 
-marshal_to_writer :: proc(w: io.Writer, v: any) -> (err: Marshal_Error) {
+marshal_to_writer :: proc(w: io.Writer, v: any, pretty: ^Pretty) -> (err: Marshal_Error) {
```

Comment:
Please don't make it require this parameter. Either make a wrapper procedure or make it default to `nil` and work without it.

Reply:
ah sorry for the late reply, i'll just use `pretty: ^Pretty = nil` and not use a bool flag inside the struct